mesonbuild/backend/ninjabackend.py

```diff
@@ -992,15 +992,14 @@ def generate_vala_compile(self, target, outfile):
             # Library name
             args += ['--library=' + target.name]
             # Outputted header
-            hname = os.path.join(self.get_target_dir(target), target.name + '.h')
+            hname = os.path.join(self.get_target_dir(target), target.vala_header)
             args += ['-H', hname]
             valac_outputs.append(hname)
             # Outputted vapi file
-            base_vapi = target.name + '.vapi'
-            vapiname = os.path.join(self.get_target_dir(target), base_vapi)
+            vapiname = os.path.join(self.get_target_dir(target), target.vala_vapi)
             # Force valac to write the vapi file in the target build dir.
             # Without this, it will write it inside c_out_dir
-            args += ['--vapi=../' + base_vapi]
+            args += ['--vapi=../' + target.vala_vapi]
```

Comment:
@nirbheek what's wrong with `get_target_dir` at this point? Isin't the C header generated in the same folder already?